### ISSUE TITLE
invalid required validation definition with custom Type

### ISSUE THREAD

**Issue by matteosuppo** [goadesign/goa#1844]
According to the docs: https://godoc.org/goa.design/goa/dsl#Payload

```
    Payload(Operands, func() {
        Required("left", "right")
    })
```

should be valid. However, with commit, I get the following error

```
invalid required validation definition: attribute must be an object (but type is operands) in attribute
```

The error is raised here: https://github.com/goadesign/goa/blob/b9d4c9b5ed81f021cf01184e6548967629e4b816/dsl/validation.go#L253

I would fix it but I'm not sure what's the expected behaviour, when it should fail and when it should not

Full design file:

```
package design

import . "goa.design/goa/http/design"
import . "goa.design/goa/http/dsl"

// API describes the global properties of the API server.
var _ = API("calc", func() {
	Title("Calculator Service")
	Description("HTTP service for adding numbers, a goa teaser")
})

// Service describes a service
var _ = Service("calc", func() {
	Description("The calc service performs operations on numbers")
	// Method describes a service method (endpoint)
	Method("add", func() {
		// Payload describes the method payload
		// Here the payload is an object that consists of two fields
		Payload(Operands, func() {
			// Both attributes must be provided when invoking "add"
			Required("a", "b")
		})
		// Result describes the method result
		// Here the result is a simple integer value
		Result(Int)
		// HTTP describes the HTTP transport mapping
		HTTP(func() {
			// Requests to the service consist of HTTP GET requests
			// The payload fields are encoded as path parameters
			GET("/add/{a}/{b}")
			// Responses use a "200 OK" HTTP status
			// The result is encoded in the response body
			Response(StatusOK)
		})
	})
})

var Operands = Type("operands", func() {
	// Attribute describes an object field
	Attribute("a", Int, "Left operand")
	Attribute("b", Int, "Right operand")
})
```

  - goa Version:  b9d4c9b5ed81f021cf01184e6548967629e4b816 (current HEAD of v2)
  - Platform (Linux/Windows etc.): Linux

**Comment by nitinmohan87** [comment]
Nice catch!
@raphael  I believe we should support this use case, right?
```
Payload(Operands, func() {
        Required("left", "right")
})
```

**Comment by raphael** [comment]
yes that was the intent :)

**Comment by raphael** [comment]
Thank you for the detailed report, this is fixed in #1918
